Calculate max stats on game launch

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

IIRC, calculating all the stats on game launch was my intention for TRX 1.0 all along… I just forgot to actually implement it. Anyway, here it is now.

The real bummer is that we already dropped storing per-level max stats in the save files. For saves created with TRX 1.0, that data simply doesn't exist anymore, so there's nothing to recover. That effectively locks us into this approach for all existing saves.

And yeah, this implementation is kinda cursed. For each level:
- We skim-load the level file, only processing objects, rooms, floor data, and items.
- We still read all injections, but ~95% get discarded. To reduce overhead, they're now loaded with a thread pool, which gives us around 40% time save.
- Then we do some duct-tape post-processing so that `carrier.c` can aggregate the pickups.
- And the cherry-on-top: we run the Lua level init, since ally setup got moved to Lua.

This code is fragile, but it sort of works. The worst part is Lua – god knows what builders stuff in there. The game is essentially running in a zombie state at that point – no textures, no anims, items unitialized, etc. And it's slow. On my machine, this adds +0.2 s to TR2 launch, and my machine's pretty damn good.

LMK what you think. You can see the calculated stats and the exact performance hit in the log by searching for `Stats_CalculateMaxStats`.

The only real alternative would be to start storing the max counters in save files again, but that still wouldn't fix existing saves made so far, since that data just isn't there anymore.

#### Testing

We need to:

- start the game, level skip to the end, and check the final stats
- load the savegame from #4432, and check the final stats
- compare the total numbers in the tables and see if I got them right (just counting the items)
- compare the total numbers in the tables against the logs / final stats screens
- × TR1, TR2, TR1UB, TR2GM
- pay extra attention to Nightmare in vegas
    - should be played when you collect all secrets
    - shouldn't be played when you fail to collect some secrets
    - collecting some secrets, save, quit, launch, load, collect remaining secrets – should still let you play it
    - final stats screen should include both Nightmare in vegas and the main game pickups/kills/secrets stats (I think this is changed by this PR, please LMK if that's the case, I'll adjust the changelog)
- allies and allied pickups in TR2 / TR2GM are not to be counted

rr- committed:
level: read max stats on game launch

Resolves #4432.

rr- committed:
inject: parallelize injection loading

rr- committed:
level/reader: do not scan max stats again

rr- committed:
stats: detach savegame dependency

aredfan commented:
> LMK what you think. You can see the calculated stats and the exact performance hit in the log by searching for `Stats_CalculateMaxStats`.

The game's performance when starting up seems okay to me. These are the figures from my logs.

- TR1 took 0.15 s
- TR2 took 0.22 s

> * final stats screen should include both Nightmare in vegas and the main game pickups/kills/secrets stats (I think this is changed by this PR, please LMK if that's the case, I'll adjust the changelog)

I can confirm this is the case although the final stats screen is shown before starting Nightmare in Vegas.

<details><summary>Screenshot</summary>

![20251220_184639_Kingdom](https://github.com/user-attachments/assets/a1f45cd5-113d-4930-a254-40555e90387d)

</details>

(Testing is in progress. 😄)

lahm86 commented:
I think performance is fine 👍 

- [x] Secret pickups in TR2 don't seem to be registering in the stats (using the console or an older save seems OK).

rr- commented:
> > LMK what you think. You can see the calculated stats and the exact performance hit in the log by searching for `Stats_CalculateMaxStats`.
> 
> The game's performance when starting up seems okay to me. These are the figures from my logs.
> 
> * TR1 took 0.15 s
> * TR2 took 0.22 s
> 
> > * final stats screen should include both Nightmare in vegas and the main game pickups/kills/secrets stats (I think this is changed by this PR, please LMK if that's the case, I'll adjust the changelog)
> 
> I can confirm this is the case although the final stats screen is shown before starting Nightmare in Vegas.

Thanks, fixed – final stats screen will include the bonus levels only when it's invoked from within a bonus level.

> I think performance is fine 👍

I'm really happy to hear that :)

> * [ ]  Secret pickups in TR2 don't seem to be registering in the stats (using the console or an older save seems OK).

Fixed this as well, though this was [a bit trickier](https://github.com/LostArtefacts/TRX/compare/d9044c5ebcf25a3705e11e0d294a4ab9cb320fb0..ca7818b299b112764be407fffe7269b5eda6f07b).

aredfan commented:
- [ ] In TR1, if I execute `/set ps1-enemies 0`, and skip through all the levels from start to finish, the total kills shows `295`. In develop, this would show `294` instead.

lahm86 reviewed (approved):
LGTM.
Re the PS mummy, I'd be in favour of just changing the config option comment to say to restart the game rather than just the level.

rr- committed:
strings: update ps1 enemies help text

rr- commented:
> LGTM. Re the PS mummy, I'd be in favour of just changing the config option comment to say to restart the game rather than just the level.

Agreed, this is the simplest fix – pushed.
